resources/web/wwi/multimedia_client.js

```diff
@@ -0,0 +1,383 @@
+/* global webots, SystemInfo */
+
+'use strict';
+
+class MultimediaClient { // eslint-disable-line no-unused-vars
+  constructor(view, parentObject, contextMenu) {
+    this.view = view;
+    this.domElement = document.createElement('img');
+    this.domElement.style.background = 'white';
+    this.domElement.id = 'remoteScene';
+    this.domElement.setAttribute('draggable', false);
+    parentObject.appendChild(this.domElement);
+
+    this.navigationEnabled = true;
+    this.contextMenu = contextMenu;
+    this.robotWindows = [];
+    this.worldInfo = {title: null, infoWindow: null};
+
+    this.mouseDown = 0;
+    this.onmousemove = (e) => { this._onMouseMove(e); };
+    this.lastMousePosition = null;
+    this.touchEvent = { move: false };
+    this.ontouchmove = (e) => { this._onTouchMove(e); };
+    this.ontouchend = (e) => { this._onTouchEnd(e); };
+  }
+
+  disconnect() {
+    this.domElement.src = '';
+  }
+
+  finalize(onready) {
+    this.domElement.addEventListener('mousedown', (e) => { this._onMouseDown(e); }, false);
+    this.domElement.addEventListener('mouseup', (e) => { this._onMouseUp(e); }, false);
+    this.domElement.addEventListener('wheel', (e) => { this._onWheel(e); }, false);
+    this.domElement.addEventListener('touchstart', (event) => { this._onTouchStart(event); }, true);
+    this.domElement.addEventListener('contextmenu', (e) => { this._onContextMenu(e); }, false);
+    if (typeof onready === 'function')
+      onready();
+  }
+
+  setWorldInfo(title, infoWindowName) {
+    this.worldInfo = {title: title, infoWindow: infoWindowName};
+  }
+
+  setRobotWindow(robotName, windowName) {
+    this.robotWindows.push([robotName, windowName]);
+  }
+
+  setFollowed(solidId, mode) {
+    var socket = this.view.stream.socket;
+    if (!socket || socket.readyState !== 1)
+      return;
+    socket.send('follow: ' + mode + ',' + solidId);
+  }
+
+  showContextMenu(object) {
+    this.contextMenu.show(object, this.lastMousePosition);
+  }
+
+  requestNewSize() {
+    if (this.lastWidth === this.domElement.width && this.lastHeight === this.domElement.height)
+      return;
+    this.view.stream.socket.send('resize: ' + this.domElement.width + 'x' + this.domElement.height);
+    this.lastWidth = this.domElement.width;
+    this.lastHeight = this.domElement.height;
+  }
+
+  resize(width, height) {
+    this.requestNewSize();
+  }
+
+  processServerMessage(data) {
+    if (data.startsWith('multimedia: ')) {
```

Comment:
This is an old message that was already implemented, and all the messages currently uses this protocol.
But in general using the colon is not causing any issues in the existing messages.
I introduced a couple of more complex messages where I had to sent a JSON objects, because when you start to send more than one string I agree that it is complicated with this plain format.
In any case the messages needs to be synchronized in Webots and in webots.min.js, so it will be in any case error prone even with JSON objects.

We can decide to move all the messages to JSON objects, but this should apply to all of them (including the ones for x3d streaming) and will be break completely the backwards compatibility.
Because if it is tricky and error prone for all the mjpeg streaming messages, than it is the same for all the other messages.